### PR TITLE
fix: Tests failed because select trigger text content changes

### PR DESCRIPTION
**Pull request by YueyingLu**
### Description

1. Move `ScreenreaderOnly` out of Trigger to fix test `createWrapper().findSelect().getElement().textContent` by customer. 
2. Treat placeholder as generic button content. It currently set to aria-labelledby only when there is selection([code ref1](https://github.com/cloudscape-design/components/blob/main/src/select/utils/use-select.ts#L167), [code ref2](https://github.com/cloudscape-design/components/blob/main/src/select/parts/trigger.tsx#L53)). As it is actually button content, it should be updated when there is selection made, and can be overriden when there is ariaLabel. 

### How has this been tested?

[_How did you test to verify your changes?_]

[_How can reviewers test these changes efficiently?_]

[_Check for unexpected visual regressions, see [`CONTRIBUTING.md`](CONTRIBUTING.md#run-visual-regression-tests) for details._]

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [ ] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/select/parts/trigger.tsx [hunk 1]
```diff
@@ -47,11 +46,7 @@ const Trigger = React.forwardRef(
     let triggerContent = null;
     if (!selectedOption) {
       triggerContent = (
-        <span
-          aria-disabled="true"
-          className={clsx(styles.placeholder, styles.trigger)}
-          id={triggerProps.ariaLabelledby}
```
Comment (YueyingLu) [inline src/select/parts/trigger.tsx:53]: This id now(without this pr) only set when there is selectedOption https://github.com/cloudscape-design/components/blob/main/src/select/utils/use-select.ts#L167 . With this pr, it behaves as button content, will be read out when no aria-label/aria-labelledby set.

File: src/select/internal.tsx [hunk 2]
```diff
@@ -143,21 +146,21 @@ const InternalSelect = React.forwardRef(
...
         disabled={disabled}
-        ariaLabel={ariaLabel}
         triggerVariant={triggerVariant}
         triggerProps={getTriggerProps(disabled)}
         selectedOption={selectedOption}
         isOpen={isOpen}
         inFilteringToken={__inFilteringToken}
         {...formFieldContext}
         controlId={controlId}
+        ariaLabelledby={[formFieldContext.ariaLabelledby, selectAriaLabelId].filter(label => !!label).join(' ')}
```
Comment (just-boris) [inline src/select/internal.tsx:156]: We have a reusable utility for such case

https://github.com/cloudscape-design/components/blob/cff14478e987a745a26066af444337e0496a2075/src/internal/utils/strings/join-strings.ts#L4-L7

Comment (YueyingLu) [reply src/select/internal.tsx]: Good point. Updated in the new revision.

File: src/select/__tests__/select-a11y.test.tsx [hunk 3]
```diff
@@ -59,4 +58,15 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
...
+
+  test('Trigger should have refer to the element using aria-label value', () => {
+    const { wrapper } = renderSelect({ selectedOption: { label: 'First', value: '1' } });
+    const ariaLabelId = wrapper
+      .findTrigger()
+      .getElement()
+      .getAttribute('aria-labelledby')
+      ?.match(/select-arialabel-.* /)?.[0]
+      .trim();
+    expect(wrapper.getElement().querySelector(`#${ariaLabelId}`)).toHaveTextContent('select');
```
[the remark this answers is not in the document]
Comment (YueyingLu) [reply src/select/__tests__/select-a11y.test.tsx]: Good idea. Updated in the new revision.

**Review by just-boris — APPROVED** [review]

**Pull request merged**